auth, variables: add description about ldap authentication (#13501)
pingcap/docs: system-variables.md

- Default value: `OFF`
- Determines whether to allow explicitly specifying the values of the column with the `AUTO_RANDOM` attribute in the `INSERT` statement.

### authentication_ldap_sasl_auth_method_name <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Enumeration
- Default value: `SCRAM-SHA-1`
- Possible values: `SCRAM-SHA-1`, `SCRAM-SHA-256`, and `GSSAPI`.
- For LDAP SASL authentication, this variable specifies the authentication method name.

### authentication_ldap_sasl_bind_base_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""
- For LDAP SASL authentication, this variable limits the search scope within the search tree. If a user is created without the `AS ...` clause, TiDB will automatically search the `dn` in LDAP server according to the user name.

### authentication_ldap_sasl_bind_root_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP SASL authentication, this variable specifies the `dn` used to login to the LDAP server to search users.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the `dn` used to log in to the LDAP server to search users.
```

- Type: Boolean
- Default value: `ON`
- This variable controls whether to use the high precision mode when computing the window functions.

### authentication_ldap_sasl_auth_method_name <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Enumeration
- Default value: `SCRAM-SHA-1`
- Possible values: `SCRAM-SHA-1`, `SCRAM-SHA-256`, `GSSAPI`.

> - For LDAP SASL authentication, the authentication method name.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the authentication method name.
```

### authentication_ldap_sasl_bind_base_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP SASL authentication, this variable limits the search scope within the search tree. If a user is created without `AS ...` clause, TiDB will automatically search the `dn` in LDAP server according to the user name.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable limits the search scope within the search tree. If a user is created without the `AS ...` clause, TiDB will automatically search the `dn` in LDAP server according to the user name.
```

### authentication_ldap_sasl_bind_root_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""
- For LDAP SASL authentication, the `dn` used to login to the LDAP server to search users.

### authentication_ldap_sasl_bind_root_pwd <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP SASL authentication, the password used to login to the LDAP server to search users.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the password used to log in to the LDAP server to search users.
```

### authentication_ldap_sasl_ca_path <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP SASL authentication, the absolute path of the certificate authority file for StartTLS connections.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the absolute path of the certificate authority file for StartTLS connections.
```

### authentication_ldap_sasl_init_pool_size <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `10`
- Range: `[1, 32767]`

> - For LDAP SASL authentication, the initial size of the pool of connections to the LDAP server.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the initial size of the pool of connections to the LDAP server.
```

### authentication_ldap_sasl_max_pool_size <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `1000`
- Range: `[1, 32767]`

> - For LDAP SASL authentication, the maximum size of the pool of connections to the LDAP server.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the maximum size of the pool of connections to the LDAP server.
```

### authentication_ldap_sasl_server_host

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""
- For LDAP SASL authentication, the LDAP server host.

### authentication_ldap_sasl_server_port

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `389`
- Range: `[1, 65535]`
- For LDAP SASL authentication, the LDAP server TCP/IP port number.

### authentication_ldap_sasl_tls

- Scope: GLOBAL
- Persists to cluster: Yes

> - Type Boolean

Reviewer 2: ```suggestion
- Type: Boolean
```

> - For LDAP SASL authentication, the LDAP server host.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the LDAP server host.
```

Reviewer: 中文里是 “LDAP Server 的主机名或地址”。

> - For LDAP SASL authentication, the LDAP server TCP/IP port number.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable specifies the LDAP server TCP/IP port number.
```

- Default value: `OFF`

> - For LDAP SASL authentication, whether connections by the plugin to the LDAP server are protected with StartTLS.

Reviewer: ```suggestion
- For LDAP SASL authentication, this variable controls whether connections by the plugin to the LDAP server are protected with StartTLS.
```

### authentication_ldap_simple_auth_method_name <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Enumeration
- Default value: `SIMPLE`
- Possible values: `SIMPLE`.

> - For LDAP simple authentication, the authentication method name. The only supported value is `SIMPLE`.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the authentication method name. The only supported value is `SIMPLE`.
```

### authentication_ldap_simple_bind_base_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP simple authentication, this variable limits the search scope within the search tree. If a user is created without `AS ...` clause, TiDB will automatically search the `dn` in LDAP server according to the user name.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable limits the search scope within the search tree. If a user is created without the `AS ...` clause, TiDB will automatically search the `dn` in LDAP server according to the user name.
```

### authentication_ldap_simple_bind_root_dn <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP simple authentication, the `dn` used to login to the LDAP server to search users.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the `dn` used to log in to the LDAP server to search users.
```

### authentication_ldap_simple_bind_root_pwd <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP simple authentication, the password used to login to the LDAP server to search users.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the password used to log in to the LDAP server to search users.
```

### authentication_ldap_simple_ca_path <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""

> - For LDAP simple authentication, the absolute path of the certificate authority file for StartTLS connection.

Reviewer 3: ```suggestion
- For LDAP simple authentication, the absolute path of the certificate authority file for StartTLS connections.
```

> - For LDAP simple authentication, the absolute path of the certificate authority file for StartTLS connections.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the absolute path of the certificate authority file for StartTLS connections.
```

### authentication_ldap_simple_init_pool_size <span class="version-mark">New in v7.1.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `10`
- Range: `[1, 32767]`

> - For LDAP simple authentication, the initial size of the pool of connections to the LDAP server.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the initial size of the pool of connections to the LDAP server.
```

### authentication_ldap_simple_max_pool_size

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `1000`
- Range: `[1, 32767]`
- For LDAP simple authentication, the maximum size of the pool of connections to the LDAP server.

### authentication_ldap_simple_server_host

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: String
- Default value: ""
- For LDAP simple authentication, the LDAP server host.

### authentication_ldap_simple_server_port

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `389`
- Range: `[1, 65535]`
- For LDAP simple authentication, the LDAP server TCP/IP port number.

### authentication_ldap_simple_tls

- Scope: GLOBAL
- Persists to cluster: Yes

> - Type Boolean

Reviewer 2: ```suggestion
- Type: Boolean
```

> - For LDAP simple authentication, the maximum size of the pool of connections to the LDAP server.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the maximum size of the pool of connections to the LDAP server.
```

> - For LDAP simple authentication, the LDAP server host.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the LDAP server host.
```

Reviewer: 中文里写的是 LDAP Server 的主机名或地址

> - For LDAP simple authentication, the LDAP server TCP/IP port number.

Reviewer: ```suggestion
- For LDAP simple authentication, this variable specifies the LDAP server TCP/IP port number.
```
